src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -964,4 +895,103 @@ public void setNumOfRubricSubQuestions(int numOfRubricSubQuestions) {
         return rubricSubQuestions;
     }
 
+    /**
+     * Class to calculate the statistics of responses for a rubric question
+     */
+    private static class RubricStatistics {
+
+        /**
+         * Last element in each row stores the total number of responses for the sub-question.
+         * e.g.
+         * responseFrequency[subQuestionIndex][choiceIndex]
+         * -> is the number of times choiceIndex is chosen for subQuestionIndex
+         * responseFrequency[subQuestionIndex][numOfRubricChoices]
+         * -> is the total number of the responses for the given sub-question
+         */
+        int[][] responseFrequency;
+
+        /**
+         * Values are set to 0 if there are no responses to that sub-question.
+         * Average value is set to 0 if there are no assigned weights.
+         * e.g.
+         * percentageFrequencyAndAverageValue[subQuestionIndex][choiceIndex]
+         * -> is the percentage choiceIndex is chosen for subQuestionIndex
+         * percentageFrequencyAndAverageValue[subQuestionIndex][numOfRubricChoices]
+         * -> is the average weight of the responses for the given sub-question
+         */
+        float[][] percentageFrequencyAndAverage;
+
+        List<FeedbackResponseAttributes> responses;
+        FeedbackRubricQuestionDetails questionDetails;
+
+        int numOfRubricSubQuestions;
+        int numOfRubricChoices;
+        int responseTotalIndex;
+
+        RubricStatistics(List<FeedbackResponseAttributes> responses, FeedbackRubricQuestionDetails questionDetails) {
+
+            this.responses = responses;
+            this.questionDetails = questionDetails;
+
+            this.numOfRubricSubQuestions = questionDetails.numOfRubricSubQuestions;
+            this.numOfRubricChoices = questionDetails.numOfRubricChoices;
+            this.responseTotalIndex = numOfRubricChoices;
+
+            calculateResponseFrequency();
+            calculatePercentageFrequencyAndAverage();
+        }
+
+        void calculateResponseFrequency() {
+            responseFrequency = new int[numOfRubricSubQuestions][numOfRubricChoices + 1];
+            // Count frequencies
+            for (FeedbackResponseAttributes response : responses) {
+                FeedbackRubricResponseDetails frd = (FeedbackRubricResponseDetails) response.getResponseDetails();
+                for (int i = 0; i < numOfRubricSubQuestions; i++) {
+                    int chosenChoice = frd.getAnswer(i);
+                    if (chosenChoice != -1) {
+                        responseFrequency[i][chosenChoice] += 1;
+                        responseFrequency[i][responseTotalIndex] += 1;
+                    }
+                }
+            }
+        }
+
+        /**
+         * Calculates the percentage frequencies for each choice and average value for each sub-question
+         * Requires responseFrequency to be computed
```

Comment:
Added a not null assertion on responseFrequency to ensure it is at least initialised (which is currently in calculateResponseFrequency).